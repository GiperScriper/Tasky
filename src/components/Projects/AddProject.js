import React, { Component } from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';

class AddProject extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isAdditing: false,
    };
  }

  handleAdditing = () => {
    this.setState({
      isAdditing: !this.state.isAdditing,
    });
  };

  render() {
    return (
      <div className="projects__item projects__item--noborder">
        {this.state.isAdditing ? (
          <form className="additing-form">
            <input type="text" placeholder="Project name" className="additing-form__input" />
            <textarea placeholder="Project description" className="additing-form__textarea" />
            <div className="projects__footer">
              <button className="button button--cancel" onClick={this.handleAdditing}>
                Cancel
              </button>
              <button className="button button--success">Save</button>
            </div>
          </form>
        ) : (
          <div className="projects__add" onClick={this.handleAdditing}>
            <FontAwesomeIcon icon={faPlus} className="projects__add-icon" />
            <div className="projects__add-title">Add Project</div>
          </div>
        )}
      </div>
    );
  }
}

export default AddProject;
