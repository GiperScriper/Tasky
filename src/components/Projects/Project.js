import React from 'react';
import { Link } from 'react-router-dom';

const Project = ({ data }) => (
  <div>
    <h1>
      <Link to={`/projects/${data.id}`}>{data.title}</Link>
    </h1>
    <p>{data.description}</p>
  </div>
);

export default Project;
