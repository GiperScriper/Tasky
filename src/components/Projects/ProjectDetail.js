import React from 'react';

const ProjectDetail = ({ match }) => (
  <div>
    <h1>Project Detail Page: {match.params.id}</h1>
  </div>
);

export default ProjectDetail;
