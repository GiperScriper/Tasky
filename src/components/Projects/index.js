import React from 'react';

const Projects = () => (
  <div>
    <h1>Projects</h1>
  </div>
);

export default Projects;
