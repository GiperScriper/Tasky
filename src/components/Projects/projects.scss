@import 'assets/styles/_vars';

.projects {
  display: flex;
  flex-wrap: wrap;
  &__item {
    display: flex;
    flex-direction: column;

    position: relative;

    width: 300px;
    height: 300px;

    margin: 0 20px 20px 0;
    padding: 10px;

    border-top-width: 2px;
    border-top-style: solid;
    box-sizing: border-box;
    border-radius: 4px;

    box-shadow: 0px 0px 10px 2px $light-gray;

    @include transition-animation;
    &:hover {
      box-shadow: 0px 0px 40px -4px $gray;
    }
    &--noborder {
      border: none;
    }
  }
  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;

    margin-bottom: 10px;
    padding-bottom: 10px;

    color: $gray;

    border-bottom: 1px solid $light-gray;
  }
  &__title {
    font-size: 1.6rem;
    color: $text-color;
  }
  &__description {
    flex: 1 1 100%;
  }
  &__footer {
    display: flex;
    justify-content: space-between;
    align-items: center;

    color: $gray;
    font-size: 0.85rem;
  }
  &__members {
    margin-left: 6px;
  }
  &__creation-time {
    margin-left: 6px;
  }
  &__add {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    flex: 1 1 100%;

    color: $gray;

    cursor: pointer;
  }
  &__add-icon {
    font-size: 3rem;
  }
  &__add-title {
    font-size: 1.8rem;

    margin-top: 10px;
  }
}

.additing-form {
  display: flex;
  flex-direction: column;
  flex: 1 1 100%;
  &__input {
    margin-bottom: 10px;
  }
  &__textarea {
    flex-grow: 1;

    margin-bottom: 10px;

    resize: none;
  }
  .button {
    flex: 1 1 50%;
    &:last-child {
      margin-left: 10px;
    }
  }
}
