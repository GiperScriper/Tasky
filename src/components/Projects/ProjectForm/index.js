import { compose, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import moment from 'moment';

import { createProject } from '@/actions/project';
import ProjectForm from './ProjectForm';
import { getRandomColor } from '@/helpers/getRandomColor';
import { colors } from '@/components/Projects/constants';

export const onSubmit = props => data => {
  const { createProject, toggleItemView, userId } = props;
  const color = getRandomColor(colors);

  createProject({
    project: {
      ...data,
      color,
      creatorId: userId,
      createdAt: moment().format(),
      members: [userId],
    },
    toggleItemView,
  });
};

const mapStateToProps = ({ auth }) => ({
  userId: auth.userId,
});

const mapDispatchToProps = dispatch => ({
  createProject: data => dispatch(createProject(data)),
});

export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withHandlers({
    onSubmit,
  })
)(ProjectForm);
