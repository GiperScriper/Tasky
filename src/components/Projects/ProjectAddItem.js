import React, { Component } from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';

import ProjectForm from '@/components/Projects/ProjectForm';

class ProjectAddItem extends Component {
  state = {
    isEnabled: false,
  };

  toggleItemView = () => {
    this.setState(prevState => ({
      isEnabled: !prevState.isEnabled,
    }));
  };

  render() {
    return (
      <div className="projects__item projects__item--noborder">
        {this.state.isEnabled ? (
          <ProjectForm toggleItemView={this.toggleItemView} />
        ) : (
          <div className="projects__add" onClick={this.toggleItemView}>
            <FontAwesomeIcon icon={faPlus} className="projects__add-icon" />
            <div className="projects__add-title">Add Project</div>
          </div>
        )}
      </div>
    );
  }
}

export default ProjectAddItem;
