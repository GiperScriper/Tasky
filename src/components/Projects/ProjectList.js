import React from 'react';
import PropTypes from 'prop-types';

import ProjectItem from './ProjectItem';
import ProjectAddItem from './ProjectAddItem';

export const ProjectList = ({ projects }) => (
  <ul className="projects">
    {projects.map(project => (
      <li key={project.id}>
        <ProjectItem data={project} />
      </li>
    ))}
    <li>
      <ProjectAddItem />
    </li>
  </ul>
);

ProjectList.defaultProps = {
  projects: [],
};

ProjectList.propTypes = {
  projects: PropTypes.array.isRequired,
};

export default ProjectList;
