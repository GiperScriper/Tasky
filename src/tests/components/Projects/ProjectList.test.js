import React from 'react';
import { shallow } from 'enzyme';
import { ProjectList } from '@/components/Projects/ProjectList';
import projects from '@/tests/fixtures/projects';

describe('<ProjectList />', () => {
  it('should render ProjectList component with projects data', () => {
    const wrapper = shallow(<ProjectList projects={projects} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ProjectList with empty data', () => {
    const wrapper = shallow(<ProjectList />);
    expect(wrapper).toMatchSnapshot();
  });
});
