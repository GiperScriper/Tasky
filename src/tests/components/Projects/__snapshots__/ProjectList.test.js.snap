// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectList /> should render ProjectList component with projects data 1`] = `
<ul
  className="projects"
>
  <li
    key="1"
  >
    <Component
      data={
        Object {
          "createdAt": "2018-08-07T17:43:05.792Z",
          "creatorId": "xdsdfgfdgfgf",
          "description": "some sample project 1",
          "id": 1,
          "title": "Sample 1",
        }
      }
    />
  </li>
  <li
    key="2"
  >
    <Component
      data={
        Object {
          "createdAt": "2018-08-07T17:43:05.792Z",
          "creatorId": "xdsdfgfdgfgf2",
          "description": "some sample project 2",
          "id": 2,
          "title": "Sample 2",
        }
      }
    />
  </li>
  <li
    key="3"
  >
    <Component
      data={
        Object {
          "createdAt": "2018-08-07T17:43:05.792Z",
          "creatorId": "xdsdfgfdgfgf3",
          "description": "some sample project 3",
          "id": 3,
          "title": "Sample 3",
        }
      }
    />
  </li>
  <li
    key="4"
  >
    <Component
      data={
        Object {
          "createdAt": "2018-08-07T17:43:05.792Z",
          "creatorId": "xdsdfgfdgfgf4",
          "description": "some sample project 4",
          "id": 4,
          "title": "Sample 4",
        }
      }
    />
  </li>
  <li>
    <ProjectAddItem />
  </li>
</ul>
`;

exports[`<ProjectList /> should render ProjectList with empty data 1`] = `
<ul
  className="projects"
>
  <li>
    <ProjectAddItem />
  </li>
</ul>
`;
